From 78ee96785053ecb3198963d0d781930a35935874 Mon Sep 17 00:00:00 2001
From: Maximilian Schmidt <maximilian.schmidt@tuhh.de>
Date: Wed, 9 Nov 2022 17:38:52 +0100
Subject: [PATCH] congatec mickledore support

---
 conf/layer.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/conf/layer.conf b/conf/layer.conf
index 60cc30c..a7825ff 100644
--- a/conf/layer.conf
+++ b/conf/layer.conf
@@ -12,4 +12,4 @@ BBFILE_PRIORITY_congatec = "7"
 LAYERDEPENDS_congatec = "intel"
 
 LAYERVERSION_congatec = "7"
-LAYERSERIES_COMPAT_congatec = "dunfell"
+LAYERSERIES_COMPAT_congatec = "mickledore"
-- 
2.30.2

