diff --git a/meta/recipes-devtools/rust/rust-cross-canadian.inc b/meta/recipes-devtools/rust/rust-cross-canadian.inc
index 17f933959f..95a6d9145a 100644
--- a/meta/recipes-devtools/rust/rust-cross-canadian.inc
+++ b/meta/recipes-devtools/rust/rust-cross-canadian.inc
@@ -78,6 +78,8 @@ do_install () {
 	export CARGO_HTTP_MULTIPLEXING=false
 
 	export CARGO_HTTP_CAINFO="\$OECORE_NATIVE_SYSROOT/etc/ssl/certs/ca-certificates.crt"
+	export LIBCLANG_PATH="\$OECORE_NATIVE_SYSROOT/usr/lib"
+	export BINDGEN_EXTRA_CLANG_ARGS_${@d.getVar('RUST_TARGET_SYS').replace('-', '_')}="--sysroot=\$OECORE_TARGET_SYSROOT"
 	EOF
 }
 
