From 8adbfcd786a435f4f7fdc73972a9111c75ec3ea9 Mon Sep 17 00:00:00 2001
From: Maximilian Schmidt <maximilian.schmidt@tuhh.de>
Date: Sat, 11 Dec 2021 17:11:26 +0100
Subject: [PATCH] SystemCallArchitecture is x86 and x86-64 for udev

---
 units/systemd-udevd.service.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/units/systemd-udevd.service.in b/units/systemd-udevd.service.in
index c146b0f7f8..4a5d495f62 100644
--- a/units/systemd-udevd.service.in
+++ b/units/systemd-udevd.service.in
@@ -37,7 +37,7 @@ RestrictRealtime=yes
 RestrictSUIDSGID=yes
 SystemCallFilter=@system-service @module @raw-io
 SystemCallErrorNumber=EPERM
-SystemCallArchitectures=native
+SystemCallArchitectures=x86-64 x86
 LockPersonality=yes
 IPAddressDeny=any
 {{SERVICE_WATCHDOG}}
