From 92121be744b368adaec8c82ff1de3bf4101c291b Mon Sep 17 00:00:00 2001
From: Maximilian Schmidt <maximilian@schmidt.so>
Date: Sun, 1 Jun 2025 14:33:57 +0000
Subject: [PATCH 2/3] hint bindgen to SDK sysroot for libclang

---
 meta/recipes-devtools/rust/cargo_1.84.1.bb | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/meta/recipes-devtools/rust/cargo_1.84.1.bb b/meta/recipes-devtools/rust/cargo_1.84.1.bb
index 9c4e4a0dd2..b54ade890d 100644
--- a/meta/recipes-devtools/rust/cargo_1.84.1.bb
+++ b/meta/recipes-devtools/rust/cargo_1.84.1.bb
@@ -62,6 +62,9 @@ do_install:append:class-nativesdk() {
 	export CARGO_HTTP_CAINFO="\$OECORE_NATIVE_SYSROOT/etc/ssl/certs/ca-certificates.crt"
 
 	export CARGO_BUILD_TARGET="\$TARGET_PREFIX"gnu
+
+	export LIBCLANG_PATH="\$OECORE_NATIVE_SYSROOT/usr/lib"
+	export BINDGEN_EXTRA_CLANG_ARGS_${@d.getVar('RUST_TARGET_SYS').replace('-', '_')}="--sysroot=\$OECORE_TARGET_SYSROOT"
 	EOF
 }
 
-- 
2.49.0

